src/ctap2/attestation.rs

```diff
@@ -1126,4 +1163,140 @@ pub mod test {
             );
         }
     }
+
+    mod hmac_secret {
+        use std::convert::TryFrom;
+
+        use crate::{
+            crypto::{
+                COSEAlgorithm, COSEEC2Key, COSEKey, COSEKeyType, CryptoError, Curve,
+                PinUvAuthProtocol, SharedSecret,
+            },
+            ctap2::{attestation::HmacSecretResponse, commands::CommandError},
+            AuthenticatorInfo,
+        };
+
+        fn make_test_secret(pin_protocol: u64) -> Result<SharedSecret, CommandError> {
+            let fake_unused_key = COSEKey {
+                alg: COSEAlgorithm::ECDH_ES_HKDF256,
+                key: COSEKeyType::EC2(COSEEC2Key {
+                    curve: Curve::SECP256R1,
+                    x: vec![],
+                    y: vec![],
+                }),
+            };
+
+            let pin_protocol = PinUvAuthProtocol::try_from(&AuthenticatorInfo {
+                pin_protocols: Some(vec![pin_protocol]),
+                ..Default::default()
+            })?;
+
+            let key = {
+                let aes_key = 0..32;
+                let hmac_key = 32..64;
+                match pin_protocol.id() {
+                    1 => aes_key.collect(),
+                    2 => hmac_key.chain(aes_key).collect(),
+                    _ => unimplemented!(),
+                }
+            };
+
+            Ok(SharedSecret::new_test(
+                pin_protocol,
+                key,
+                fake_unused_key.clone(),
+                fake_unused_key,
+            ))
+        }
+
+        const PIN_PROTOCOL_2_IV: [u8; 16] = [0; 16]; // PIN protocol 1 uses a hard-coded all-zero IV
+        const EXPECTED_OUTPUT1: [u8; 32] = [
```

Comment:
If possible, add a comment here how one can verify that this is the expected output (link to spec, if it's in there, or external commands used to calculate it).